### DISCUSSION TITLE
Me Again - TriggerListeners ILogger not injecting

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1673] [Q&A] [answered]
I'm trying to create a trigger listener but for some reason ILogger is not being injected and wants me to pass in the ILogger

Snippet from my class.

```csharp
public class UpdateTaskDetailLogListener : TriggerListenerSupport
{
    private readonly ILogger<UpdateTaskDetailLogListener> logger;
    //private readonly string exampleValue;
    //private readonly ApplicationDbContext _context;
    
    public UpdateTaskDetailLogListener(ILogger<UpdateTaskDetailLogListener> logger)
    {
        this.logger = logger;
        //this.exampleValue = exampleValue;
        //_context = context;
    }
```

When I do `ITriggerListener _l = new UpdateTaskDetailLogListener();` I get an error about not passing the ILogger.

I thought it is supposed to be injected.

Thanks

**Comment by lahma** [comment, marked as answer]
You shouldn't instantiate the trigger listener yourself, but use the callback in `AddQuartz`, the example has this:

https://github.com/quartznet/quartznet/blob/6c81befdb55bda55b6dbc596630c2a19c1aef973/src/Quartz.Examples.AspNetCore/Startup.cs#L183

**Reply by CrazyMoka** [reply]
So I should only add my triggers in the startup?
when would I use:
scheduler.ListenerManager.AddTriggerListener(UpdateTaskDetailLogListener); // because this doesn't work either.

**Reply by lahma** [reply]
When you need dependency injection in v3, you need to use startup-style configuration.

**Reply by CrazyMoka** [reply]
Thank you
